test-app/tests/unit/shared-adapter-behaviour.js

```diff
@@ -1149,19 +1149,18 @@ SharedBehavior.mockCreateTests = function () {
       });
   });
 
-  test('fails when match args not present in createRecord attributes', async function (assert) {
-    let mock = mockCreate('profile').match({
+  test('does not call mock when match args not present in createRecord attributes', async function (assert) {
+    const baseMock = mockCreate('profile');
+    let otherMock = mockCreate('profile').match({
       description: 'correct description',
     });
 
     await FactoryGuy.store
       .createRecord('profile', { description: 'wrong description' })
-      .save()
-      .catch(() => {
-        assert.ok(true);
-        // our mock was NOT called
-        assert.strictEqual(mock.timesCalled, 0);
-      });
+      .save();
+
+    assert.strictEqual(baseMock.timesCalled, 1);
+    assert.strictEqual(otherMock.timesCalled, 0);
```

Comment:
use of `.catch()` no longer catches the errors on these when the handler misses. I couldn't figure out a way to catch the errors thrown there

Ultimately it's because the request handler function is now async, and Pretender treats async handlers differently - it doesn't throw, `erroredRequest` doesn't get called
https://github.com/pretenderjs/pretender/blob/65fbf0a608f2e4b4fd79657a77a41f52077074a9/src/pretender.ts#L238-L242

Instead, I've just made sure to assert `timesCalled` is as expected